oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/ItemImpl.java

```diff
@@ -319,8 +303,39 @@ public String toString() {
 
     //-----------------------------------------------------------< internal >---
     @NotNull
-    String getOakName(String name) throws RepositoryException {
-        return sessionContext.getOakName(name);
+    String getOakName() throws RepositoryException {
+        return perform(new ItemOperation<String>(dlg, "getName") {
```

Comment:
@reschke Do you know why this needs the locking? IMHO the dlg.name is a final field so I don't think this requires any locking.

Reply:
Nope.